### PR TITLE
Fix for https://github.com/OpenFeign/feign/issues/462

### PR DESCRIPTION
**Pull request by maximdim**

### CODE REVIEW

File: core/src/main/java/feign/Logger.java [hunk 1]
```diff
@@ -89,7 +90,9 @@ protected Response logAndRebufferResponse(String configKey, Level logLevel, Resp
       }
 
       int bodyLength = 0;
-      if (response.body() != null) {
+      if (response.body() != null && !(status == 204 || status == 205)) {
```
Comment (adriancole) [inline core/src/main/java/feign/Logger.java]: According to the rules below, it seems erroneous if there a body when status like this is sent? what http client are you using? could you add a MockWebServer test to show this? Why don't we fix the client instead?

Comment (maximdim) [inline core/src/main/java/feign/Logger.java]: I'm using OkHttpClient. There was similar issue in retrofit and from what I can tell they decided that this is not okhttp bug: https://github.com/square/retrofit/issues/735

**Comment by adriancole** [comment]
We can make different decisions than retrofit (we have in the past). please
add tests to the base http client. you may find this only affects okhttp,
or not.

On Tue, Sep 20, 2016 at 9:41 AM, Dmitri Maximovich <notifications@github.com

**Comment by adriancole** [comment]
since this only affects the logger, I'm not really bothered by this. we should add a test that it works. Can you do this?

**Pull request merged**